.config/waybar/scratchpad-indicator.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+output="$(swaymsg -t get_tree | jq -r 'recurse(.nodes[]?)|recurse(.floating_nodes[]?)|select(.name == "__i3_scratch").floating_nodes[]|"`<b>"+.name+"`</b> - "+.app_id+" "+(.id|tostring)')"
+number="$(swaymsg -r -t get_tree | jq -r 'recurse(.nodes[]) | first(select(.name=="__i3_scratch")) | .floating_nodes | length')"
+
+if [ "$number" -gt 0 ]; then
```

Comment:
The scratchpad indicator won't show up when $number==0 -> text =="", meaning that no right-click "move to scratchpad" action is possible when there are no windows in the scratchpad already

Reply:
Hi! @FilippoBonazziSUSE I am struggling to understand and to reproduce what you describe. Could you be so kind as to explain step by step what you do and where the script fails?

Reply:
Hi @taschenlampe, I was referring to the fact that the scratchpad indicator has a right-click action to move the active window to the scratchpad. However, when the scratchpad is empty, the scratchpad indicator won't show up (by design), so this right-click context action is not accessible.

The right-click context action is nice for user discoverability, but if it doesn't show up in the default state it's not discoverable :smile: .

I personally don't use the right-click action, so it doesn't bother me. In the end it's a trade-off between space efficiency (don't show the scratchpad indicator when empty) and a nice-to-have feature (right click indicator to move window to scratchpad).

Reply:
@FilippoBonazziSUSE  Oh I see now!
 My thought: "New" users may not know what a scratchpad is and will be confused of having something in the bar indicator that mostly will be unused. So I´d say, it's a nice little feature for the ones that know about scratchpad, they'll enjoy this little detail :-)

Reply:
Sure! Let's go ahead with it then